### PR TITLE
Azure - Add Azure container registry resource

### PR DESCRIPTION
**Pull request by wesyao**
- Add new acr resource
- Add `azure-mgmt-containerregistry` dependency
- Add tests, provision templates, and test cassette

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/resources/container_registry.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 Capital One Services, LLC
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/container_registry.py]: nit: 2019 :)

Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/container_registry.py]: nit: 2019 :)

**Comment by stefangordon** [comment]
@wesyao have you signed the CLA? https://docs.google.com/forms/d/e/1FAIpQLSfwtl1s6KmpLhCY6CjiY8nFZshDwf_wrmNYx1ahpsNFXXmHKw/viewform

**Comment by wesyao** [comment]
@stefangordon yeah, i believe i have, but i'll do it again to make sure

**Pull request merged**